apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -312,11 +317,11 @@ export class ChangeAction extends BaseRushAction {
   }
 
   private _getChangedPackageNames(): string[] {
-    const changedFolders: (string | undefined)[] | undefined = VersionControl.getChangedFolders(
+    const changedFiles: string[] = VersionControl.getChangedFiles(
       this._targetBranch,
       this._noFetchParameter.value
     );
-    if (!changedFolders) {
+    if (changedFiles.length === 0) {
```

Comment:
I think this was a logic error because of the nullable return type for `changeFolders` mentioned above. I suspect this optimisation was never hitting since `changedFolders` would always be truthy, even with no changes (`![] === false`). This should now skip the (potentially) expensive change-checking if no git changes are detected.